src/esp/io/URDFParser.cpp

```diff
@@ -0,0 +1,1058 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+// Code adapted from Bullet3/examples/Importers/ImportURDFDemo ...
+
+#include <Corrade/Utility/DebugStl.h>
+#include <Corrade/Utility/Directory.h>
+#include <Corrade/Utility/String.h>
+#include <Magnum/Math/Quaternion.h>
+#include <iostream>
+
+#include "URDFParser.h"
+
+#include "tinyxml2/tinyxml2.h"
+
+// using namespace tinyxml2;
+using tinyxml2::XMLDocument;
+using tinyxml2::XMLElement;
+
+namespace Cr = Corrade;
+namespace Mn = Magnum;
+
+namespace esp {
+namespace io {
+namespace URDF {
+
+bool Parser::parseURDF(std::shared_ptr<Model>& urdfModel,
+                       const std::string& filename) {
+  Mn::Debug silence{logMessages ? &std::cout : nullptr};
+
+  // override the previous model with a fresh one
+  urdfModel = std::make_shared<Model>();
+  sourceFilePath_ = filename;
+  urdfModel->m_sourceFile = filename;
+
+  std::string xmlString = Corrade::Utility::Directory::readString(filename);
+
+  XMLDocument xml_doc;
+  xml_doc.Parse(xmlString.c_str());
+  if (xml_doc.Error()) {
+    Mn::Debug{}
+        << "Parser::parseURDF - XML parse error, aborting URDF parse/load.";
```

Comment:
How about print the name of the file that failed to parse? And maybe this is Mn::Warn or Mn::Error, not Mn::Debug?